scene/resources/bit_map.cpp

```diff
@@ -32,13 +32,20 @@
 
 #include "core/io/image_loader.h"
 
-void BitMap::create(const Size2 &p_size) {
+void BitMap::create(const Size2i &p_size) {
 	ERR_FAIL_COND(p_size.width < 1);
 	ERR_FAIL_COND(p_size.height < 1);
 
-	width = p_size.width;
-	height = p_size.height;
-	bitmask.resize((((width * height) - 1) / 8) + 1);
+	int w = static_cast<int>(p_size.width);
+	int h = static_cast<int>(p_size.height);
+	ERR_FAIL_COND(static_cast<int64_t>(w) * static_cast<int64_t>(h) > INT32_MAX);
+
+	Error err = bitmask.resize(((w * h) / 8) + 1);
```

Comment:
For `w * h == 8k` it should `resize(k)`, not `resize(k + 1)`, see #48549.

But as I've mentioned in #48545 there's still an issue with `BitMap` using `Vector` for the storage as it doubles the allocated memory when resizing. In case of `BitMap` I think it makes sense to use a container which resizes to the exact needed memory. There's a kinda related discussion about such containers in #64120.

Reply:
Reapplied your fix. We could use `TightLocalVector` but I would rather wait for #64204 (or a similar solution) getting merged to avoid unnecessary work.

Reply:
Indeed, no need to change it in this PR. Also I wasn't suggesting specifically `TightLocalVector` as I'm not even sure if it fits this use case (like maybe `Vector` is currently used because of COW or something like that, I don't know). Just wanted to point out this existing issue. Maybe we should reopen #48545 to not forget about it?